docs/design/release_automation_design.md

```diff
@@ -0,0 +1,72 @@
+## Release Automation Process Design
+
+### Abstract
+
+The release automation process is an automated system that will handle Github
+releases. What this entails is tagging a release, ensure that the release blurb
+contains all the changes that are to go out, and potentially handle release
+artifacts.
+
+### Specification
+
+The release automation process will be triggered by a periodic AWS Lambda
+(Lambda) job that will signify whether or not regeneration of the SDK needs to
+occur and/or tag a Github release.
+
+Regeneration of the SDK will be determined by diffing the swagger.yaml files in
+the SDK and Firecracker. If there is a difference in the swagger files, this
+will regenerate the SDK and also tag a new release. In the event that there are
+no differences, the SDK will check some sort of notifier that may trigger a
+newly tagged release. An example notifier could be a certain change in the
+CHANGELOG.md file which would flag for a release in Amazon DynamoDB (DynamoDB).
+By storing a release flag in a database allows for multiple different ways for
+triggering a release and allows for swapping the type of release notifier.
+
+The newly generated SDK will need to be tested before deployment or before
+tagging a release, which will use the GitLab test runner to do the testing.
+This will require some modifications on how code is deployed, meaning a
+separate branch.  Once that branch has been committed to the release process
+would need to create a PR and poll for success of the GitLab test runner.
+
+### Deployment
+
+CDK and AWS CloudFormation templates are an option, but this writing this stack
+will be short work in the AWS SDK for Go. The release process stack can be
+deployed to a new stack by simply running its make command with the given set
+of AWS credentials for the new stack. This will setup the proper DynamoDB
+tables, Lambda functions, IAM roles, and any other resources needed.
+
+### Manual Release Process
+
+Ensure that no PRs, commits, or other releases are in the process during the
```

Comment:
In the ECS agent and other related projects (like the cred helper), we make use of staging branches and pull requests.  This means that we know a commit SHA ahead of time and can use that for the release instead of requiring that other developers freeze their activities.